background/services/internal-ethereum-provider/index.ts

```diff
@@ -419,16 +442,20 @@ export default class InternalEthereumProviderService extends BaseService<Events>
         data: transactionRequest.input,
         ...transactionRequest,
         gasLimit: transactionRequest.gas, // convert gas -> gasLimit
+        // Etherjs rejects Rootstock checksummed addresses so convert to lowercase
+        from: isRootstock
+          ? transactionRequest.from &&
+            normalizeHexAddress(transactionRequest.from)
+          : transactionRequest.from,
```

Comment:
```suggestion
        from: isRootstock && transactionRequest.from !== undefined
          ? normalizeHexAddress(transactionRequest.from)
          : transactionRequest.from,
```

Let's move the existence check for `from` to the top here.

Reply:
Really hate that we have to handle `from` here when we validate its existence further down… To be tackled another day.

Reply:
✅